test/phoenix_live_view/tag_engine_test.exs

```diff
@@ -0,0 +1,83 @@
+defmodule Phoenix.LiveView.TagEngineTest do
+  use ExUnit.Case, async: true
+
+  defp to_ast(source) do
+    options = [
+      engine: Phoenix.LiveView.TagEngine,
+      file: __ENV__.file,
+      line: __ENV__.line,
+      caller: __ENV__,
+      indentation: 0,
+      source: source,
+      tag_handler: Phoenix.LiveView.HTMLEngine
+    ]
+
+    EEx.compile_string(source, options)
+  end
+
+  defmacrop keyed_comprehension(id, vars_changed) do
```

Comment:
I am honestly not a big fan of this test, it is a unit test and it can fail if we change implementation details and everything continues working. Let's discard this and add some proper tests instead (I assume in html_engine_test.exs) that asserts it returns a component tree and then we render them.

Reply:
Although I guess this test is good for testing the variable behaviour... it would be very hard otherwise, so let's keep it. :) But we should definitely not rely on it much!